content/docs/reference/design-docs/train-sim-v3/driver-behavior-modules.en.md

```diff
@@ -3,36 +3,82 @@ title: "Driver behavior modules"
 weight: 30
 ---
 
-### Driver behavior modules
 
-```mermaid
-flowchart TD
+## Design specs
 
-subgraph BuilderInputs["builder inputs"]
-    SlowdownCoefficient[slowdown coefficient]
-    PathProps[path properties]
-end
+### General pitch
 
-subgraph CallInputs["call inputs"]
-    ConcreteDrivingInstructions[concrete driving instruction]
-    CurrentTrainState[current train state]
-end
+Driver behavior modules are responsible for making driving decisions. 
+It's main responsibility, given the state of the train and an instruction, is to react to the instruction.
+This reaction is expressed as a new train state.
 
-DriverBehaviorModule[driver behavior module]
-Decision[decision]
+To perform this critical task, it needs access to additional context:
 
-BuilderInputs --> DriverBehaviorModule
-CallInputs --> DriverBehaviorModule --> Decision
+- the physical properties of the path, which are used to make coasting decisions, and to model natural forces.
+- a slowdown coefficient, which is used to adjust how much the train is slowed down compared to a 
+full power simulation.
 
-```
+The driver behavior modules are supposed to have different implementations, which would interpret the slow down coefficient differently.
 
-Driver behavior modules are responsible for making driving decisions. It's main responsibility, given the state of the train,
-is to provide a driving instruction (a braking or traction force). To perform this critical task, it needs access to additional context:
 
-- the physical properties of the path, which are used to make coasting decisions
-- driving instructions
+### API
 
-## Design decisions
+One driver behavior module is instantiated per driving instruction.
+It takes at initialization:
+- a slowdown coefficient
+- the driving instruction
+- the path properties
+
+It has two public methods:
+- `enact_decision(current_state: TrainState, t: float) -> (TrainState, float)`
+
+  Which returns what the next train state would be if there was only this one instruction to follow, and the time delta
+  to reach this state.
+
+- `truncate_integration_step(current_state: TrainState, potential_state: TrainState, t: float, dt: float) -> (TrainState, float)`
+
+  Which returns a state and time delta which respects the instruction, and is as close as possible to the potential state. 
```

Comment:
Do we have an idea how this works, how this is implemented? What does the driver behavior module do with `potential_state`?

Reply:
This is for sure something we did not discuss in our workshops. 

For the simple case illustrated with a time-speed diagram, I'd say the implementation would be simply intersecting the speed limit and the [(t, current_state) : (t + dt), (potential_state)] segment. 

Regarding driver behavior modules handling pantographs I don't really know.

I can add something to the warning section, do you have any suggestions ?

Reply:
Ah, yes: DBMs may interpret driving instructions in such way that they have an internal speed limit, different than the one in the driving instruction (in order to lose some time for example)

Your question is pertinent, but I don't know how to integrate this detail in the doc. Should I maybe add an FAQ ? Or maybe when explaining step 4 I can mention that it could be an internal speed limit as an example ?

Reply:
Maybe something to add under "design decisions"? Maybe something among the lines of:

### Should `truncate_integration_step` depend on the driver behavior module?

Yes: DBMs may interpret driving instructions in such way that they have an internal speed limit, different from the one in the driving instruction. For instance, when passed a driving instruction with a speed limit of 60km/h, a DBM wishing to loose time may reduce the speed to 50 km/h.

Reply:
Yup, that's a neat idea, thanks